### ISSUE TITLE
Additional args to `img`? (build-args, env, etc.)

### ISSUE THREAD

**Issue by ChrisPenner** [concourse/builder-task#3]
Hello! At @wireapp we're trying to cut down our docker build times in concourse and think this builder may help; I have it rigged up in my pipeline; but I need to pass additional arguments at build time such as:

- `--env blah=val` 
- `--build-args blah=val`

Taking a quick look at the `build` script looks as though there isn't a way to do this currently; 

Am I missing a way to do this? If not; do you have plans to implement this, or a suggestion for how you'd like it to be done so I can submit a Pull request? Thanks!

Cheers 🍻

**Comment by ChrisPenner** [comment]
Incidentally it looks like I can actually pass additional args as part of `CONTEXT` which isn't expanded inside quotes e.g. `"$CONTEXT"`; but that's probably not a great idea haha.

**Comment by vito** [comment]
@ChrisPenner Ha, yeah probably not a good idea.

How about a PR to just add `"$@"` to the args here?:

https://github.com/concourse/builder/blob/f1c9f1770069c5b09af0c07df316d0e14d51fff4/build#L83

That way any args passed to `build` would just be forwarded along.

**Comment by cirocosta** [comment]
I was just about to propose adding `--target` and `--build-args`, but indeed, perhaps allowing any flags is more ideal as @vito mentioned, giving more flexibility to folks use whatever they want from `img`. 

However, this would be pretty much https://github.com/concourse/concourse/wiki/Anti-Patterns#kitchen-sink-config, wouldn't it?

**Comment by ChrisPenner** [comment]
ATM all I really need is (multiple) --build-arg and --env; but I suspect I might have other needs as time goes on. I'm currently held up by bugs in build-kit; but apparently those are fixed; so I'll take another look at this once there's a release available.

**Comment by yasn77** [comment]
Hi :wave: 

As I don't really see any objections to this feature, I have opened a PR to add this ability in #12. I agree with @cirocosta, that rather than having a 'generic' option to pass any option, it is better to be specific. If there is a need to include other options, I think they should just be added to params.

I would really appreciate this getting looked at, I also have a requirement for this and don't particularly want to maintain my own fork :smiley: 

Thanks

**Comment by yasn77** [comment]
After submitting a PR that I hadn't properly tested, I have now opened a new PR #13. This works as expected and allows you to specify build args. I agree with the point made by @cirocosta regarding it being an anti-pattern to have a 'catch all' option that gets passed through. If new options are required, I think it is better to update the build script to support them.

**Comment by vito** [comment]
fixed by https://github.com/concourse/builder-task/pull/13 - thanks! I'll build and publish a new version shortly.
